include/eve/module/core/regular/impl/simd/x86/sub.hpp

```diff
@@ -8,48 +8,142 @@
+      else if constexpr( match(c,category::int16x8 , category::uint16x8 ) ) return _mm_mask_sub_epi16   (src, m, v, w);
+      else if constexpr( match(c,category::int8x64 , category::uint8x64 ) ) return _mm512_mask_sub_epi8 (src, m, v, w);
+      else if constexpr( match(c,category::int8x32 , category::uint8x32 ) ) return _mm256_mask_sub_epi8 (src, m, v, w);
+      else if constexpr( match(c,category::int8x16 , category::uint8x16 ) ) return _mm_mask_sub_epi8    (src, m, v, w);
+      else                                                                  return sub.behavior(cpu_{}, opts, v, w);
+    }
   }
-}
 
 }
+
+// #include <eve/concept/value.hpp>
+// #include <eve/detail/implementation.hpp>
+
+// #include <type_traits>
+
+// namespace eve::detail
+// {
+// // -----------------------------------------------------------------------------------------------
+// // Masked case
+// template<conditional_expr C, arithmetic_scalar_value T, typename N>
+// EVE_FORCEINLINE wide<T, N>
+//                 sub_(EVE_SUPPORTS(sse2_),
+//                      C const                         &cx,
+//                      wide<T, N> const                &v,
+//                      wide<T, N> const                &w) noexcept requires x86_abi<abi_t<T, N>>
+// {
+//   constexpr auto c = categorize<wide<T, N>>();
+
+//   if constexpr( C::is_complete || abi_t<T, N>::is_wide_logical )
+//   {
+//     return sub_(EVE_RETARGET(cpu_), cx, v, w);
+//   }
+//   else
+//   {
+//     auto src = alternative(cx, v, as<wide<T, N>> {});
+//     auto m   = expand_mask(cx, as<wide<T, N>> {}).storage().value;
+
+//     if constexpr( c == category::float32x16 ) return _mm512_mask_sub_ps(src, m, v, w);
+//     else if constexpr( c == category::float64x8 ) return _mm512_mask_sub_pd(src, m, v, w);
+//     else if constexpr( c == category::int8x64 ) return _mm512_mask_sub_epi8(src, m, v, w);
+//     else if constexpr( c == category::int8x32 ) return _mm256_mask_sub_epi8(src, m, v, w);
+//     else if constexpr( c == category::int8x16 ) return _mm_mask_sub_epi8(src, m, v, w);
+//     else if constexpr( c == category::int16x32 ) return _mm512_mask_sub_epi16(src, m, v, w);
+//     else if constexpr( c == category::int16x16 ) return _mm256_mask_sub_epi16(src, m, v, w);
+//     else if constexpr( c == category::int16x8 ) return _mm_mask_sub_epi16(src, m, v, w);
+//     else if constexpr( c == category::int32x16 ) return _mm512_mask_sub_epi32(src, m, v, w);
+//     else if constexpr( c == category::int32x8 ) return _mm256_mask_sub_epi32(src, m, v, w);
+//     else if constexpr( c == category::int32x4 ) return _mm_mask_sub_epi32(src, m, v, w);
+//     else if constexpr( c == category::int64x8 ) return _mm512_mask_sub_epi64(src, m, v, w);
+//     else if constexpr( c == category::int64x4 ) return _mm256_mask_sub_epi64(src, m, v, w);
+//     else if constexpr( c == category::int64x2 ) return _mm_mask_sub_epi64(src, m, v, w);
+//     else return sub_(EVE_RETARGET(cpu_), cx, v, w);
```

Comment:
what's this?

Reply:
Somethign I forgot to cleanup from the previous code